jck/crosscheck/crosscheck.sh

```diff
@@ -0,0 +1,225 @@
+#!/usr/bin/env bash
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+genTargetList() {
+	inputFile=$TEST_ROOT/TKG/parallelList.mk
+	outputFile=$listOfExistingTargets
+	rm -rf $outputFile
+	
+	# If the TKG generated target list already exists don't generate it again
+	if [ ! -f "$inputFile" ] ; then
+		if [ ! -d "$TEST_ROOT/TKG" ] ; then
+			echo "Can't find TKG under $TEST_ROOT/TKG"
+			echo "Please ensure setup was done properly"
+			exit 1
+		else 
+			echo "Generating list of existing playlist targets.." 
+			cd $TEST_ROOT/TKG
+			make genParallelList NUM_MACHINES=1 TEST=jck
```

Comment:
@renfeiw is there a way to directly call genParallelList java code only? Similar to https://github.com/AdoptOpenJDK/openjdk-tests/blob/ff27e6d37252092c43f7a9983b310e46e18c8d81/system/daaLoadTest/playlist.xml#L9

The above code compiles TKG and runs machine info again.
```
00:06:30.887  ===============================================
00:06:30.887  Running test jck_crosscheck_0 ...
00:06:30.887  ===============================================
00:06:30.887  jck_crosscheck_0 Start Time: Thu Nov 19 13:46:16 2020 Epoch Time (ms): 1605822376096
00:06:30.887  "/home/jenkins/workspace/Grinder/openjdkbinary/j2sdk-image/bin/java" -Xshareclasses:destroyAll; "/home/jenkins/workspace/Grinder/openjdkbinary/j2sdk-image/bin/java" -Xshareclasses:groupAccess,destroyAll; echo "cache cleanup done";
00:06:30.887  
00:06:30.887  Attempting to destroy all caches in cacheDir /home/jenkins/javasharedresources/
00:06:30.887  
00:06:30.887  JVMSHRC806I Compressed references persistent shared cache "sharedcc_jenkins" has been destroyed. Use option -Xnocompressedrefs if you want to destroy a non-compressed references cache.
00:06:30.887  JVMSHRC005I No shared class caches available
00:06:30.887  cache cleanup done
00:06:30.887  variation: NoOptions
00:06:30.887  JVM_OPTIONS:  
00:06:30.887  { itercnt=1; \
00:06:30.887  mkdir -p "/home/jenkins/workspace/Grinder/openjdk-tests/TKG/../TKG/test_output_16058223755291/jck_crosscheck_0"; \
00:06:30.887  cd "/home/jenkins/workspace/Grinder/openjdk-tests/TKG/../TKG/test_output_16058223755291/jck_crosscheck_0"; \
00:06:30.887  /home/jenkins/workspace/Grinder/openjdk-tests/TKG/../jck/crosscheck/crosscheck.sh jck11 /home/jenkins/workspace/Grinder/openjdk-tests/TKG/../../../../jck_root/JCK11-unzipped /home/jenkins/workspace/Grinder/openjdk-tests/TKG/..; \
00:06:30.887  if [ $? -eq 0 ] ; then echo ""; echo "jck_crosscheck_0""_PASSED"; echo ""; cd /home/jenkins/workspace/Grinder/openjdk-tests/TKG/..;  else echo ""; echo "jck_crosscheck_0""_FAILED"; echo ""; fi; } 2>&1 | tee -a "/home/jenkins/workspace/Grinder/openjdk-tests/TKG/../TKG/test_output_16058223755291/TestTargetResult";
00:06:30.887  Thu Nov 19 13:46:16 PST 2020
00:06:30.887  Generating list of existing playlist targets..
00:06:30.887  make[5]: Entering directory '/home/jenkins/workspace/Grinder/openjdk-tests/TKG'
00:06:30.887  JAVA_HOME is set to /home/jenkins/workspace/Grinder/openjdkbinary/j2sdk-image
00:06:30.887  rm -f -r /home/jenkins/workspace/Grinder/openjdk-tests/TKG/../TKG/test_output_compilation; \
00:06:30.887  mkdir -p /home/jenkins/workspace/Grinder/openjdk-tests/TKG/../TKG/test_output_compilation; \
00:06:30.887  (ant -f ./scripts/build_tools.xml -DTEST_JDK_HOME=/home/jenkins/workspace/Grinder/openjdkbinary/j2sdk-image -DTEST_ROOT=/home/jenkins/workspace/Grinder/openjdk-tests/TKG/.. 2>&1; echo $? ) | tee /home/jenkins/workspace/Grinder/openjdk-tests/TKG/../TKG/test_output_compilation/compilation.log; \
00:06:30.888  if [ -z $(tail -1 /home/jenkins/workspace/Grinder/openjdk-tests/TKG/../TKG/test_output_compilation/compilation.log | grep 0) ]; then perl scripts/moveDmp.pl --compileLogPath=/home/jenkins/workspace/Grinder/openjdk-tests/TKG/../TKG/test_output_compilation/compilation.log --testRoot=/home/jenkins/workspace/Grinder/openjdk-tests/TKG/..; false; else rm -f -r /home/jenkins/workspace/Grinder/openjdk-tests/TKG/../TKG/test_output_compilation; fi
00:06:31.879  Buildfile: /home/jenkins/workspace/Grinder/openjdk-tests/TKG/scripts/build_tools.xml
00:06:32.866  
00:06:32.866  build:
00:06:32.866  
00:06:32.866  clean:
00:06:32.866     [delete] Deleting: /home/jenkins/workspace/Grinder/openjdk-tests/TKG/autoGenEnv.mk
00:06:32.866     [delete] Deleting: /home/jenkins/workspace/Grinder/openjdk-tests/TKG/utils.mk
00:06:32.866     [delete] Deleting directory /home/jenkins/workspace/Grinder/openjdk-tests/TKG/bin
00:06:32.866  
00:06:32.866  init:
00:06:32.866      [mkdir] Created dir: /home/jenkins/workspace/Grinder/openjdk-tests/TKG/bin
00:06:32.866  
00:06:32.866  getDependentLibs:
00:06:32.866       [exec] --------------------------------------------
00:06:32.866       [exec] path is set to /home/jenkins/workspace/Grinder/openjdk-tests/TKG/../TKG/lib
00:06:32.866       [exec] task is set to default
00:06:32.866       [exec] dependencyList is set to json_simple
00:06:32.866       [exec] --------------------------------------------
00:06:32.866       [exec] Starting download third party dependent jars
00:06:32.866       [exec] --------------------------------------------
00:06:32.866       [exec] downloading dependent third party jars to /home/jenkins/workspace/Grinder/openjdk-tests/TKG/../TKG/lib
00:06:32.866       [exec] /home/jenkins/workspace/Grinder/openjdk-tests/TKG/../TKG/lib/json-simple.jar exists with correct hash, not downloading
00:06:32.866       [exec] downloaded dependent third party jars successfully
00:06:32.866  
...
```

Reply:
Yes, but it is a bit complicated to use:
```${TEST_JDK_HOME}/bin/java -cp $(Q)$(TKG_JAR)$(P)$(JSON_SIMPLE)$(Q) org.testKitGen.MainRunner --mode=parallelList --spec=$(SPEC) --jdkVersion=$(JDK_VERSION) --impl=$(JDK_IMPL) --buildList=${BUILD_LIST}  --testTarget=$(TESTTARGET) --testList=$(TESTLIST) --numOfMachines=$(NUM_MACHINES) --testTime=$(TEST_TIME) --TRSSURL=$(TRSS_URL)```

Reply:
@llxia - looks like we need to make all those additional parameters available to the script in order to issue the command above. Is it okay to leave this part as is, e.g. allow machine info to run again for the  `jck_target_folder_check` target (which would runs once per each jck extended build).

Reply:
I am ok to run it using the make command (run TKG compile and Machine info) as it will not take too much time.

Just for the record, in case this is needed in the future. We may already have the values. Also, some of the values can be skipped: 
```
$(JAVA_COMMAND) -cp $(Q)$(TKG_JAR)$(P)$(JSON_SIMPLE)$(Q) org.testKitGen.MainRunner --mode=parallelList --spec=$(SPEC) --jdkVersion=$(JDK_VERSION) --impl=$(JDK_IMPL) --buildList=jck --testTarget=jck -numOfMachines=1
```